src/core/Akka/IO/UdpListener.cs

```diff
@@ -118,6 +118,12 @@ private void DoReceive(SocketAsyncEventArgs e, IActorRef handler)
         {
             try
             {
+                if (e.SocketError != SocketError.Success)
```

Comment:
Can you move this closer to the source of the error?
In this case, this should probably moved to the `if (!Socket.ReceiveFromAsync(e))` line inside the `ReceiveAsync()` method.

Reply:
Should also note that the original Scala code simply throws when any socket error occured. We should do the same and let the socket actor parent/guardian decide what to do with the failing socket actor.

Reply:
By closing the udp socket actor it will be vulnerable to different denial of service attacks. In my case the issue was caused by the underlying network just telling the **receiver** socket that his **sender** socket had a problem reaching a udp client. Underneath this was an ICMP packet sent from the client to the server _"Dear sender listening on port x the port you are trying to reach isn't available - stop sending data"_. This is an external hint and no unrecoverable internal error.

Would you mind providing a link to the original akka scala source? I didn't find it quickly within the original akka udp implementation.

Thanks in advance and best regards

Reply:
The problem is that `System.Net.Sockets.SocketError` does not just report ICMP error, it also reports a host of other error as outlined [here](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.socketerror?view=net-5.0), trying to address all that in a robust way will be quite a challange.

The related Scala code is [here](https://github.com/akka/akka/blob/master/akka-actor/src/main/scala/akka/io/UdpListener.scala#L107-L111), as you can see, it is not trying to handle any errors, it just does a simple try...finally and release the buffer it borrowed from a buffer pool.